app/cdap/components/NamespaceAdmin/SourceControlManagement/reducer.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { ISourceControlManagement, ISourceControlManagementAuth, scmAuthType } from './types';
+
+export const defaultSourceControlManagement: ISourceControlManagement = {
+  provider: 'GITHUB',
+  link: '',
+  defaultBranch: '',
+  pathPrefix: '',
+  auth: {
+    type: scmAuthType.pat,
+    tokenName: '',
+    token: '',
+  },
+};
+
+export const sourceControlManagementFormReducer = (state: ISourceControlManagement, action) => {
+  let newAuth: ISourceControlManagementAuth;
+  switch (action.type) {
+    case 'SET_PROVIDER':
+      return {
```

Comment:
seems like this reducer could be a lot smaller if you did something like 
```
case 'SET_VALUE': 
  return {
  ...state,
  [action.value.key]: action.value.value,
  };
  ```
  and then a different one for auth?

Reply:
addressed